provider/awssd/aws_sd.go

```diff
@@ -264,6 +316,220 @@ func (p *AWSSDProvider) updatesToCreates(changes *plan.Changes) (creates []*endp
 	return creates, deletes
 }
 
+func (p *AWSSDProvider) handleSRVChanges(namespaces []*sd.NamespaceSummary, creates []*endpoint.Endpoint, deletes []*endpoint.Endpoint) ([]*endpoint.Endpoint, []*endpoint.Endpoint, error) {
```

Comment:
Could this function be broken down into smaller pieces, please? It feels like all the magic happens here in `200` lines and it's a bit difficult to follow. It would be nice to have a description of what this function returns.

Reply:
broken into 3 logical pieces